cluster-operator/src/main/java/io/strimzi/operator/cluster/model/CruiseControl.java

```diff
@@ -419,42 +411,77 @@ protected List<EnvVar> getEnvVars() {
-        data.put(API_USER_PASSWORD_KEY, Util.encodeToBase64(apiUserPassword));
-        data.put(API_AUTH_FILE_KEY, Util.encodeToBase64(authCredentialsFile));
-
-        return data;
+    public static Map<String, String> generateCruiseControlApiCredentials(PasswordGenerator passwordGenerator,
+                                                                          Secret oldSecret,
+                                                                          CruiseControlUser adminUser) {
+        if (oldSecret != null) {
+            // The credentials should not change with every release
+            // So if the secret with credentials already exists, we re-use the values
+            // But we use the new secret to update labels etc. if needed
+            var data = oldSecret.getData();
+            var adminPassword = data.get(API_ADMIN_PASSWORD_KEY);
+            var userPassword = data.get(API_USER_PASSWORD_KEY);
+            var authFile = data.get(API_AUTH_FILE_KEY);
+            if (adminPassword == null || adminPassword.isBlank() || userPassword == null
+                    || userPassword.isBlank() || authFile == null || authFile.isBlank()) {
+                throw new RuntimeException(format("Secret %s is invalid", oldSecret.getMetadata().getName()));
+            } else if (!Util.decodeFromBase64(authFile).contains(API_ADMIN_NAME) || !Util.decodeFromBase64(authFile).contains(API_USER_NAME)) {
+                throw new RuntimeException(format("Secret %s has invalid authentication file", oldSecret.getMetadata().getName()));
+            } else {
+                return data;
+            }
+        } else {
+            String apiAdminPassword = passwordGenerator.generate();
+            String apiUserPassword = passwordGenerator.generate();
+
+            /*
+             * Create Cruise Control API auth credentials file following Jetty's
+             *  HashLoginService's file format: username: password [,rolename ...]
+             */
+            String authCredentialsFile =
+                API_ADMIN_NAME + ": " + apiAdminPassword + "," + API_ADMIN_ROLE + "\n" + 
+                    API_USER_NAME + ": " + apiUserPassword + "," + API_USER_ROLE + "\n";
+
+            if (adminUser != null) {
+                authCredentialsFile += adminUser.username() + ": " + adminUser.password() + "," + API_ADMIN_ROLE + "\n";
+            }
+
+            Map<String, String> data = new HashMap<>(3);
+            data.put(API_ADMIN_PASSWORD_KEY, Util.encodeToBase64(apiAdminPassword));
+            data.put(API_USER_PASSWORD_KEY, Util.encodeToBase64(apiUserPassword));
+            data.put(API_AUTH_FILE_KEY, Util.encodeToBase64(authCredentialsFile));
+
+            return data;
+        }
     }
 
+    /**
+     * Cruise Control user credentials.
+     * @param username Username.
+     * @param password Password.
+     */
+    public record CruiseControlUser(String username, String password) { }
```

Comment:
Perhaps it would be after to override `toString()` so as to mask the password, so that it's safe to use a `CruiseControlUser` in logging. Even if we don't use it in logging today, it's the sort of thing which might get changed in the future and inadvertently result in logging sensitive information.

Reply:
That's a good idea. Thanks.